src/saltext/vmware/states/vm.py

```diff
@@ -88,3 +88,125 @@ def set_boot_manager(
+        service_instance = connect.get_service_instance(opts=__opts__, pillar=__pillar__)
+    ret = {"name": name, "changes": {}, "result": True, "comment": ""}
+    if datacenter_name:
+        dc_ref = utils_common.get_mor_by_property(service_instance, vim.Datacenter, datacenter_name)
+        vm_ref = utils_common.get_mor_by_property(
+            service_instance, vim.VirtualMachine, name, "name", dc_ref
+        )
+    else:
+        vm_ref = utils_common.get_mor_by_property(service_instance, vim.VirtualMachine, name)
+
+    if utils_vm.get_snapshot(vm_ref, snapshot_name, None):
+        ret["comment"] = "snapshot was already created"
+        return ret
+
+    if __opts__["test"]:
+        ret["changes"]["new"] = f"Snapshot {snapshot_name} will be created."
+        ret["comment"] = "These options are set to change."
+        return ret
+
+    snapshot = utils_vm.create_snapshot(vm_ref, snapshot_name, description, memory, quiesce)
+
+    if isinstance(snapshot, vim.vm.Snapshot):
+        ret["changes"]["new"] = f"Snapshot {snapshot_name} created."
+        ret["comment"] = "created"
+        return ret
+    else:
+        ret["comment"] = "failed to create"
+        return ret
+
+
+def snapshot_absent(
+    name,
+    snapshot_name,
+    snapshot_id=None,
+    remove_children=False,
+    datacenter_name=None,
+    service_instance=None,
+):
+    """
+    Create virtual machine snapshot.
+
+    name
+        The name of the virtual machine.
+
+    snapshot_name
+        The name for the snapshot being destroyed. Not unique
+
+    snapshot_id
+        (optional) ID of snapshot to be destroyed.
+
+    remove_children
+        (optional, Bool) Remove snapshots below snapshot being removed in tree.
+
+    datacenter_name
+        (optional) The name of the datacenter containing the virtual machine.
+
+    service_instance
+        (optional) The Service Instance from which to obtain managed object references.
+    """
+    if service_instance is None:
+        service_instance = connect.get_service_instance(opts=__opts__, pillar=__pillar__)
+    ret = {"name": name, "changes": {}, "result": True, "comment": ""}
+    if datacenter_name:
+        dc_ref = utils_common.get_mor_by_property(service_instance, vim.Datacenter, datacenter_name)
+        vm_ref = utils_common.get_mor_by_property(
+            service_instance, vim.VirtualMachine, name, "name", dc_ref
+        )
+    else:
+        vm_ref = utils_common.get_mor_by_property(service_instance, vim.VirtualMachine, name)
+
+    snapshot = utils_vm.get_snapshot(vm_ref, snapshot_name, snapshot_id)
+
+    if snapshot is None:
+        ret["comment"] = f"Snapshot {snapshot_name} doesn't exist"
+        return ret
+
+    if __opts__["test"]:
+        ret["changes"]["new"] = f"Snapshot {snapshot_name} will be destroyed."
+        ret["comment"] = "These options are set to change."
+        return ret
+
+    utils_vm.destroy_snapshot(snapshot.snapshot, remove_children)
```

Comment:
**question** can this fail? We're never returning a result of False :thinking:

Reply:
Yes, it will error if it can not destroy the snap.

Reply:
When doing salt states (or modules for that matter) - we shouldn't let exceptions percolate: we should capture them and report something useful to the user.

There's probably a more nuanced view, but generally speaking we should handle errors. For example:

```
local:
----------
          ID: blerp
    Function: neato.kaboom
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/home/wayne/programming/salt/salt/state.py", line 2191, in call
                  ret = self.states[cdata["full"]](
                File "/home/wayne/programming/salt/salt/loader/lazy.py", line 149, in __call__
                  return self.loader.run(run_func, *args, **kwargs)
                File "/home/wayne/programming/salt/salt/loader/lazy.py", line 1201, in run
                  return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
                File "/home/wayne/programming/salt/salt/loader/lazy.py", line 1216, in _run_as
                  return _func_or_method(*args, **kwargs)
                File "/home/wayne/programming/salt/salt/loader/lazy.py", line 1249, in wrapper
                  return f(*args, **kwargs)
                File "/home/wayne/salt/var/cache/salt/minion/extmods/states/neato.py", line 16, in kaboom
                  raise Exception('oops')
              Exception: oops
     Started: 17:16:43.886980
    Duration: 1.227 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   1.227 ms
```

That's useful for a developer, but not so much for devops.

```
local:
----------
          ID: blerp
    Function: neato.less_boom
      Result: False
     Comment: Oh dear that was unexpected!
     Started: 17:19:51.625339
    Duration: 0.503 ms
     Changes:

Summary for local
------------
Succeeded: 0
Failed:    1
------------
Total states run:     1
Total run time:   0.503 ms
```

would be better.

Basically, the comment should be anything that they need to know to solve the problem.